tests/basicScenarioTest.m

```diff
@@ -0,0 +1,153 @@
+            %verify scheduling
+            for i=2:testCase.Config.Runtime.numRounds
+                for j=1:testCase.Config.MacroEnb.numPRBs 
+                    testCase.verifyEqual(testCase.Simulation.Results.schedule(i,1,j).UeId, 1); %Assert that UE 1 is scheduled all the time
+                    testCase.verifyEqual(testCase.Simulation.Results.schedule(i,1,j).NDI, true); %Asssert that New Data Indicator indicates that there are more data.
+                    %testCase.verifyTrue(testCase.Simulation.Results.schedule(i,1,j).ModOrd == 2 ||...
+                    %testCase.Simulation.Results.schedule(i,1,j).ModOrd == 4 ||...
+                    %testCase.Simulation.Results.schedule(i,1,j).ModOrd == 6 ); %TODO: examine how to assert this
+                    %testCase.verifyEqual(testCase.Simulation.Results.schedule(i,1,j).Mcs > 20); %TODO: examine how to assert this
+                end
+            end
+            %verify HARQ and ARQ
+            arrayfun(@(x) testCase.verifyEqual(x ,0), testCase.Simulation.Results.harqRtx); %Should be 0. With good signal strength retransmission should be 0
+            arrayfun(@(x) testCase.verifyEqual(x ,0), testCase.Simulation.Results.arqRtx); %TODO: verify this statement
+            %verify BER and BLER
+            testCase.verifyTrue(mean(testCase.Simulation.Results.ber(2:end)) < 0.15 ); %Should be less than a certain threshhold. With good signal strength retransmission should be 0
+            testCase.verifyTrue(mean(testCase.Simulation.Results.bler(2:end))< 0.15 ); %TODO: verify this statement
+            %verify CQI
+            arrayfun(@(x) testCase.verifyTrue(10 <= x && x <= 15) , testCase.Simulation.Results.cqi); %TODO: find a more narrow range and confirm
+            %Verify SNR and SINR. With only 1 Enb they should be the same
+            testCase.verifyTrue( any(abs(testCase.Simulation.Results.snrdB - testCase.Simulation.Results.sinrdB) < 1e-4 ));
+            %TODO: find a more appropiate range and/or veriy current
+            testCase.verifyTrue( 35 < mean(testCase.Simulation.Results.snrdB) && mean(testCase.Simulation.Results.snrdB) < 45 );
+            %Verify difference between pre and post Evm
+            testCase.verifyTrue(any(testCase.Simulation.Results.preEvm > testCase.Simulation.Results.postEvm));
+            %Verify throughput
+            testCase.verifyTrue( any( testCase.Simulation.Results.throughput(2:end) > 1e8) ); %With 50 PRBs the throughput is expected to be in the 100mb/s range
+            %Verify receivedPowerdBm
+            testCase.verifyTrue( any(-70 < testCase.Simulation.Results.receivedPowerdBm) && any( testCase.Simulation.Results.receivedPowerdBm < -55) ); %TODO: narrow this range and verify
+            %Verify rsrqdB, rsrpdBm, rssidBm
+            testCase.verifyTrue( any(-100 < testCase.Simulation.Results.rsrqdB) && any(testCase.Simulation.Results.rsrqdB < 0) ); %TODO: narrow this range
```

Comment:
I think this is a super idea to validate. I'm not so sure the MATLAB toolbox does what it is supposed to. Maybe create a utility function that is capable of calculating the theoretical RSRP, RSRQ and RSSI based on NDLRB e.g. This would also be super useful for simulations.

Reply:
The RSRQ are calculated as a function of RSSI, RSRP and NDLRB. So the verification of RSRQ is redundant if we use the RSSI and RSRP from the measurements. As far as i have understood it is possible to calculate a RSRP value from the NDLRB and the RSSI, but the RSSI value should be obtained from measurements. And in that case the calculations would in theory be based off of the same measurements that the result is derived from in any case. Or am I getting it wrong?

Reply:
It's not redundant if we validate the RSRP from the RSSI, that's my point. We should not use the reference measurement function for computing the RSRP, but instead derive the RSRP according to the theory (power is split over the resource grid given the received power, thus RSRP is a function of the resource grid with a given received power.)